qutebrowser/mainwindow/tabwidget.py

```diff
@@ -165,22 +165,29 @@ def get_tab_fields(self, idx):
         fields['perc_raw'] = tab.progress()
         fields['backend'] = objects.backend.name
         fields['private'] = ' [Private Mode] ' if tab.private else ''
+        
+        url = self.tab_url(idx)
```

Comment:
Dang, I didn't consider that the move to here also changes the error handling :wink:

Something like this (untested) should work:

```python
try:
    url = self.tab_url(idx)
except qtutils.QtValueError:
    fields['host'] = ''
    fields['current_url'] = ''
    fields['protocol'] = ''
else:
    fields['host'] = url.host()
    fields['current_url'] = url.toDisplayString()
    fields['protocol'] = url.scheme()
```

(the `'current_url'` one used to be `url.url()`, but `.toDisplayString()` is actually the better thing to do here).

Hope you don't mind the scope of this being a bit bigger than we originally thought!

Reply:
Haha no worries! I'll make sure to test this code before I commit it this time.